### ISSUE TITLE
Glide will print Exception when load webp image

### ISSUE THREAD

**Issue by Demo-H** [bumptech/glide#4451]
** Error log start **
2020-12-22 19:35:06.819 20362-20482/com.example.testapp W/ExifInterface: Invalid image: ExifInterface got an unsupported image format file(ExifInterface supports JPEG and some RAW image formats only) or a corrupted JPEG file to ExifInterface.
    java.io.IOException: Invalid byte order: 5249
        at android.media.ExifInterface.readByteOrder(ExifInterface.java:3134)
        at android.media.ExifInterface.isOrfFormat(ExifInterface.java:2449)
        at android.media.ExifInterface.getMimeType(ExifInterface.java:2327)
        at android.media.ExifInterface.loadAttributes(ExifInterface.java:1755)
        at android.media.ExifInterface.<init>(ExifInterface.java:1449)
        at com.bumptech.glide.load.resource.bitmap.ExifInterfaceImageHeaderParser.getOrientation(ExifInterfaceImageHeaderParser.java:40)
        at com.bumptech.glide.load.ImageHeaderParserUtils.getOrientation(ImageHeaderParserUtils.java:92)
        at com.bumptech.glide.load.resource.bitmap.Downsampler.decodeFromWrappedStreams(Downsampler.java:269)
        at com.bumptech.glide.load.resource.bitmap.Downsampler.decode(Downsampler.java:224)
        at com.bumptech.glide.load.resource.bitmap.Downsampler.decode(Downsampler.java:173)
        at com.bumptech.glide.load.resource.bitmap.ByteBufferBitmapDecoder.decode(ByteBufferBitmapDecoder.java:31)
        at com.bumptech.glide.load.resource.bitmap.ByteBufferBitmapDecoder.decode(ByteBufferBitmapDecoder.java:14)
        at com.bumptech.glide.load.engine.DecodePath.decodeResourceWithList(DecodePath.java:92)
        at com.bumptech.glide.load.engine.DecodePath.decodeResource(DecodePath.java:70)
        at com.bumptech.glide.load.engine.DecodePath.decode(DecodePath.java:59)
        at com.bumptech.glide.load.engine.LoadPath.loadWithExceptionList(LoadPath.java:76)
        at com.bumptech.glide.load.engine.LoadPath.load(LoadPath.java:57)
        at com.bumptech.glide.load.engine.DecodeJob.runLoadPath(DecodeJob.java:524)
        at com.bumptech.glide.load.engine.DecodeJob.decodeFromFetcher(DecodeJob.java:488)
        at com.bumptech.glide.load.engine.DecodeJob.decodeFromData(DecodeJob.java:474)
        at com.bumptech.glide.load.engine.DecodeJob.decodeFromRetrievedData(DecodeJob.java:426)
        at com.bumptech.glide.load.engine.DecodeJob.onDataFetcherReady(DecodeJob.java:390)
        at com.bumptech.glide.load.engine.SourceGenerator.onDataFetcherReady(SourceGenerator.java:146)
        at com.bumptech.glide.load.engine.DataCacheGenerator.onDataReady(DataCacheGenerator.java:94)
        at com.bumptech.glide.load.model.ByteBufferFileLoader$ByteBufferFetcher.loadData(ByteBufferFileLoader.java:70)
        at com.bumptech.glide.load.engine.DataCacheGenerator.startNext(DataCacheGenerator.java:74)
        at com.bumptech.glide.load.engine.SourceGenerator.startNext(SourceGenerator.java:50)
        at com.bumptech.glide.load.engine.DecodeJob.runGenerators(DecodeJob.java:310)
        at com.bumptech.glide.load.engine.DecodeJob.runWrapped(DecodeJob.java:279)
        at com.bumptech.glide.load.engine.DecodeJob.run(DecodeJob.java:234)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
        at com.bumptech.glide.load.engine.executor.GlideExecutor$DefaultThreadFactory$1.run(GlideExecutor.java:431)
**Error log end**

**Code**
        val url = "https://isparta.github.io/compare-webp/image/png_webp/webp_lossless/1.webp"
        Glide.with(this).load(url).into(imageView)
**Code**

**Glide Version**: 4.10.0
**Integration libraries**: no
**Device/Android Version**: Vivo iQOO3
**Issue**: When I load webp image, It will print exception log as above，but can load success. I don't understand why have this exception .

**Comment by sjudd** [comment]
Unfortunately this comes from Android's ExifInterface class, so we can't suppress it.

**Comment by aaronKueng** [comment]
Actually， The bug looks like comes from Android ExifInterface , but it is a bug of Glide, and happened in Webp image.
1, Why the exception happened ?
Before the exception happened,  I debug the code and find that  the image mime type check result is "unknown".
I found that ,  the ExifInterfaceImageHeaderParser get wrong header data which like this " [70, 70, -86, 8, 0, 0, 87, 69, 66, 80, 86, ......." . 
Webp file starts with "RIFFxxxxWebpxxxx",  must start with "RIFF" four characters , but the start characters begin with "FF", and "RI"  is lost. 

2, Why "FF" and not "RIFF"
First, I check the cache file,  and the header start with "RIFF" , so it is not the problem of image file.
Then, check the code of getOrientation function , We know that there are two image header parsers do the work: 
    DefaultImageHeaderParser
    ExifInterfaceImageHeaderParser
Two parsers share one image file input stream object.
The first parser DefaultImageHeaderParser call imageReader obj to get getUInt16, and check image number;
Because it is webp file, so it is not DefaultImageHeaderParser's work, so return unknown and the newt parser work.

While ExifInterfaceImageHeaderParser is working, but it does not do reset for the shared stream obj.
Because, the first parser read one UInt16, 2 bytes, so the second parser read header buffer from position = 2, not  0.
So, The begin 2 characters lost. 
Based on :
RIFF Header - 32 bits representing the ASCII characters ‘R’ ‘I’ ‘F’ ‘F’
File Size - 32 bits (uint32) representing the size of the file in bytes starting at offset 8. The maximum value of this field is 2^32 minus 10 bytes and thus the size of the whole file is at most 4GiB minus 2 bytes.
‘WEBP’ - 32 bits representing the ASCII characters ‘W’ ‘E’ ‘B’ ‘P’

We know that, webp file type check, the first four characters are the key point.  So the mime check failed.
Then, exif info check run into unknown type file type, and throw the exception:
"com.example.testapp W/ExifInterface: Invalid image: ExifInterface got an unsupported image format file(ExifInterface supports JPEG and some RAW image formats only) or a corrupted JPEG file to ExifInterface. "

Glide provide wrong stream byte data is the root cause, and not the problem of ExifInterface.
Of course, It is based on if you refer ExifInterface version > 1.2.0.
Because ExifInterface support webp header parse start 1.2.0, maybe,  I found webp parsing code in 1.2.0 and not check earlier version.
